src/Glpi/Controller/DropdownController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Routing\Attribute\Route;
+
+final readonly class DropdownController implements Controller
+{
+    #[Route("/dropdown/form/{class}", name: "glpi_dropdown_form", options: ["is_form" => true])]
+    #[Route("/dropdown/{class}", name: "glpi_dropdown")]
+    public function __invoke(Request $request): Response
+    {
+        $class = $request->attributes->getString('class');
+
+        if (!$class) {
+            throw new BadRequestException('The "class" attribute is mandatory for dropdown routes.');
+        }
+
+        if (!\is_subclass_of($class, \CommonDropdown::class)) {
+            throw new BadRequestException('The "class" attribute is mandatory for dropdown routes.');
+        }
+
+        $is_form = $request->attributes->getBoolean('is_form');
+
+        return new StreamedResponse(static function () use ($class, $is_form) {
+            dd($class, $is_form);
+
+            $dropdown = new $class();
+
+            if ($is_form) {
+                include GLPI_ROOT . "/front/dropdown.common.form.php";
+            } else {
+                include GLPI_ROOT . "/front/dropdown.common.php";
+            }
```

Comment:
We should put these files contents in the current controller, and replace the content of these files by a call to `Toolbox::deprecated()` and the execution of the current controller. It should permit to remove the legacy code from the `/front` directory and, in the same time, prevent any BC-break for plugins.

Reply:
Will do 👍